.github/workflows/ci.yaml

```diff
@@ -15,21 +15,82 @@ env:
   NODE_OPTIONS: --max_old_space_size=6144
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+permissions: {}
+
 jobs:
+  dedupe:
```

Comment:
Why is this even needed?
Why is this not solved by dependabot or even yarn?
I find it strange that every project that relies on dependabot would need some workaround like this?

This also breaks the auto rebase feature that dependabot does as it refuses to change a PR that someone else has messed with, which means that we would need to comment `@dependabot recreate` for all PRs.
This is not great from a resource perspective (we should also add concurrency with cancel to the CI workflow but that's a separate topic.)

Reply:
> Why is this even needed? Why is this not solved by dependabot or even yarn?

Yarn certainly can't control what dependabot does, but yeah I totally agree dependabot should handle it.  Hopefully this would be temporary.  Here's the issue: https://github.com/dependabot/dependabot-core/issues/5830


> This also breaks the auto rebase feature that dependabot does as it refuses to change a PR that someone else has messed with, which means that we would need to comment `@dependabot recreate` for all PRs. 

Yeah I agree, although not "all" PRs will need the deduplication step.  Theoretically, there would be less and less commits required as dependencies are brought up to date.

>This is not great from a resource perspective (we should also add concurrency with cancel to the CI workflow but that's a separate topic.)

I was going to add that but didn't want to go off-topic.

Reply:
I don't see the direct need for it to run and push in this workflow at all.

We are trying to do all kinds of weird things to make this happen. IMHO we can just make this a maintenance task, or let a task run every day or week on a schedule to clean this up.

Reply:
The existing CI checks for duplicates via `yarn dedupe --check`

Reply:
If thats all that this is really for, we can just remove that check and do this as general maintenance and/or on a separate schedule?

Reply:
I don't think that's wise.  The check is run to minimize what packages get bundled and sent to the browser.  Leaving duplicates can affect frontend results.  It's not just maintenance.

Reply:
If we make a separate action, that check if a dedupe is needed and creates a PR if it is and run that daily, it might be an easier fix? Or even run it after a merge of a dependabot PR.

We can still have the dedupe check in CI for manual PRs and skip it for dependabot?

Reply:
I personally wouldn't be a fan of that because you'd be degrading any validation and testing on the original bump, only to make more work for yourself on the subsequent deduplication.  It would also be arguably more work than this change, especially to automate opening the PR.

I honestly didn't expect push back over "skipped" messages, but since this is getting no love, what's the problem with making a separate workflow (only to run on dependabot branches) and using a GitHub app to authenticate?

I could setup a new app and transfer ownership to HA (or use an existing bot?).  Then use a [3rd party action like this one](https://github.com/marketplace/actions/github-app-token) to authenticate and commit as the app.  Someone above my pay grade just needs to add the credentials as secrets.

The only downside to that approach is that if the app commits, then CI runs twice and wastes resources.

Reply:
Here's an example from a recent PR to support keeping the check (and moving it up before the others):

https://github.com/home-assistant/frontend/actions/runs/4022152029

The CI failed during the types check, but the actual problem was the imports getting mangled with a duplicate.  Simply running the dedupe was the fix.